src/RewardsDistributor.sol

```diff
@@ -76,7 +82,14 @@ contract RewardsDistributor is IRewardDistributor {
         uint256 systemPrecision = 10 ** 18;
         uint256 distributorPrecision = 10 ** IERC20(payoutToken).decimals();
         uint256 adjustedAmount = (payoutAmount_ / systemPrecision) * distributorPrecision;
+
+        if (adjustedAmount > rewardsAmount) {
+            revert NotEnoughRewardsLeft(adjustedAmount, rewardsAmount);
```

Comment:
We need to track amount that has been distributed because at any given moment we need to know how much we still need to distribute (unclaimed rewards, or timelocked)

Then we check that new distribution value + outstanding awards do not exceed current balance.

Reply:
if we changed the behavior of the core system to instead *replace* the existing distribution (rather than adding to it), would it remove the need to do this in your analysis?

Reply:
I don't think it is correct to replace distribution. We can add many distributions based on dates and that should be possible if I understand that correctly